openraft/Cargo.toml

```diff
@@ -42,6 +42,8 @@ tracing-subscriber = { version = "0.3.3",  features=["env-filter"] }
 
 
 [features]
+default = ["serde"]
+
```

Comment:
hi @schreter , thank you for the kind review. I think we can disable `serde` by default. Then use [serde_test](https://crates.io/crates/serde_test) to do some tests.

What do you think?

Reply:
> I think we can disable serde by default. Then use [serde_test](https://crates.io/crates/serde_test) to do some tests.

Of course, *not* using `serde` by default and only activating it if the `serde` feature is on is the other (possibly preferable) option, but I'm not sure what is the best. That should be decided by one of the maintainers.

Independent of that, you still need to create an extra crate to test it with the feature enabled, then. It is to my knowledge simply impossible to test both with a feature enabled and a feature disabled inside of the same crate.

@drmingdrmer Opinions/directions?

Reply:
@schreter @PsiACE 
Disabling `serde`  by default looks good to me.

> Independent of that, you still need to create an extra crate to test it with the feature enabled, then. It is to my knowledge simply impossible to test both with a feature enabled and a feature disabled inside of the same crate.

I'm used to writing feature flag related unit tests like this:
```rust
#[cfg(feature = "backtrace")]
#[test]
fn test_backtrace() -> anyhow::Result<()> {
    let got = backtrace_str().expect("no none");
    assert!(got.contains("test_backtrace"));
    Ok(())
}

#[cfg(not(feature = "backtrace"))]
#[test]
fn test_backtrace() -> anyhow::Result<()> {
    let got = backtrace_str();
    assert!(got.is_none());
    Ok(())
}
```

And let CI run it twice: with such a feature flag on and off. 

Good enough?

Reply:
> Good enough?

Sure, it just then takes double resources in the CI :-). But if there are sufficient resources, then this indeed is the best approach.

Reply:
> Good enough?

I've written this a lot in the past, and I think it's totally feasible.